### DISCUSSION TITLE
Sortable with removable items and with layout animations

### DISCUSSION THREAD

**Discussion by joseDaKing** [clauderic/dnd-kit#684] [Q&A] [answered]
I want to have animate items that are deleted, like in the Sortable -> Vertical -> Removable example, I have tried to use animateLayoutChanges prop in the useSortable hook.

```ts
// code inside component
const sortableProps = useSortable({ 
  id: value,
  animateLayoutChanges: defaultAnimateLayoutChanges
});
```

**Comment by clauderic** [comment, marked as answer]
Hey @joseDaKing, 

There are two pieces required to have @dnd-kit manage layout animations for you, as can be seen in the removable items example.

First, you must pass a custom `measuring` config to `<DndContext>` to tell @dnd-kit to always measure droppable containers, even when not dragging. By default, @dnd-kit only measures items while dragging.

```tsx
import {MeasuringStrategy} from '@dnd-kit/core';

const measuringConfig = {
  droppable: {
    strategy: MeasuringStrategy.Always,
  },
};

<DndContext measuring={measuringConfig} />
```

Next, you need to tell `useSortable` to animate layout changes when not sorting:

```ts
import {defaultAnimateLayoutChanges} from '@dnd-kit/sortable';

function customAnimateLayoutChanges(args) {
  if (args.isSorting || args.wasDragging) {
    return defaultAnimateLayoutChanges(args);
  }
  
  return true;
}


const sortableProps = useSortable({
  animateLayoutChanges: customAnimateLayoutChanges
});
```

**Reply by joseDaKing** [reply]
@clauderic  Tried this out but it doesn't work, here is a [link to codesandbox](https://codesandbox.io/s/focused-pond-9xiv3r?file=/src/App.tsx)

**Comment by joseDaKing** [comment]
I fixed the reason was accidentally used useSortable inside dragoverlay
